src/template-tags/tickets.php

```diff
@@ -109,6 +109,164 @@ function tribe_events_count_available_tickets( $event = null ) {
 	}
 }//end if
 
+if ( ! function_exists( 'tribe_events_count_tickets_and_rsvp' ) ) {
+
+	/**
+	 * Returns Ticket and RSVP Count for an Event
+	 *
+	 * @param $event_id
+	 *
+	 * @return array
+	 */
+	function tribe_events_count_tickets_and_rsvp( $event_id ) {
+
+		$tickets = Tribe__Tickets__Tickets::get_all_event_tickets( $event_id );
+
+		$types['tickets'] = array(
+			'count'     => 0, // count of tickets currently for sale
+			'stock'     => 0, // current stock of tickets available for sale
+			'global'    => 0, // global stock ticket
+			'unlimited' => 0, // unlimited stock tickets
+			'available' => 0, // are tickets available for sale right now
+		);
+		$types['rsvp']    = array(
+			'count'     => 0,
+			'stock'     => 0,
+			'unlimited' => 0,
+			'available' => 0,
+		);
+
+		foreach ( $tickets as $ticket ) {
+
+			// If a ticket is not current for sale do not count it
+			if ( ! tribe_events_ticket_is_on_sale( $ticket ) ) {
+				continue;
+			}
+
+			// if ticket and not rsvp add to ticket array
+			if ( 'Tribe__Tickets__RSVP' !== $ticket->provider_class ) {
+				$types['tickets']['count'] ++;
+
+				if ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 0 === $types['tickets']['global'] ) {
+					$types['tickets']['global'] ++;
+				} elseif ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 1 === $types['tickets']['global'] ) {
+					continue;
+				}
+
+				$types['tickets']['stock'] = $types['tickets']['stock'] + $ticket->stock;
+
+				if ( 0 !== $types['tickets']['stock'] ) {
+					$types['tickets']['available'] ++;
+				}
+
+				if ( ! $ticket->manage_stock() ) {
+					$types['tickets']['unlimited'] ++;
+					$types['tickets']['available'] ++;
+				}
+			} else {
+				$types['rsvp']['count'] ++;
+
+				$types['rsvp']['stock'] = $types['rsvp']['stock'] + $ticket->stock;
+				if ( 0 !== $types['rsvp']['stock'] ) {
+					$types['rsvp']['available'] ++;
+				}
+
+				if ( ! $ticket->manage_stock() ) {
+					$types['rsvp']['unlimited'] ++;
+					$types['rsvp']['available'] ++;
+				}
+			}
+		}
+
+		return $types;
+	}
+}//end if
+
+
+if ( ! function_exists( 'tribe_events_display_count_and_ticket_button' ) ) {
+
+	/**
+	 * Echos Remaining Ticket Count and Purchase Buttons for an Event
+	 *
+	 * @return string/null
+	 */
+	function tribe_events_display_count_and_ticket_button() {
+
+		$event_id = get_the_ID();
+
+		// get an array for ticket and rsvp counts
+		$types = tribe_events_count_tickets_and_rsvp( $event_id );
+
+		// If we have tickets or RSVP, but everything is Sold Out then display the Sold Out messages
+		if ( ( $types['tickets']['count'] || $types['rsvp']['count'] ) && ( ! $types['tickets']['available'] && ! $types['rsvp']['available'] ) ) {
```

Comment:
This `if` is long and hard to read on one line. Consider splitting it into multiple lines:

```php
if (
  (
    $types['tickets']['count']
    || $types['rsvp']['count']
  )
  && ! $types['tickets']['available']
  && ! $types['rsvp']['available']
) {
```